onnxruntime/core/providers/migraphx/migraphx_provider_factory.cc

```diff
@@ -145,6 +145,19 @@ struct MIGraphX_Provider : Provider {
   }
 
   void Initialize() override {
+#ifdef _WIN32
+    HMODULE module = nullptr;
+    if(GetModuleHandleEx(GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS |
+                              GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT,
+                          static_cast<LPCSTR>(static_cast<void*>(InitializeRegistry)),
+                          &module) != 0) {
+      char buffer[MAX_PATH];
```

Comment:
Always use wide Windows APIs when possible.

Reply:
Should this be split between windows/linux builds then?

Reply:
I mean when a Windows API has an ANSI version and Wide version, you need to use the Wide version.
The buffer here should be std::wstring.